.claude/Agents.md

```diff
@@ -0,0 +1,267 @@
+    const record = useRecordContext();
+    return (
+        <TextField
+            {...props}  // Pass through all additional props
+            value={record?.[source]}
+        />
+    );
+};
+```
+
+### File Organization
+
+- **Feature-based structure** within packages (not type-based)
+- **One file per component** except for components that can't be used in standalone (e.g. `DataTable.Col`)
+- **Co-location** - Tests (`.spec.tsx`) and stories (`.stories.tsx`) next to components
+- **Index exports** - Each directory has an index.ts exporting public API
+- **Flat structure** within features - avoid unnecessary nesting
+
+### JSDoc
+
+Every hook or component must have basic usage described in JSDoc, including a non-trivial example. No need to document all options - IDEs display TypeScript types for that. 
+
+### Documentation
+
+Every new feature or API change must be documented. The documentation consists of Markdown files located in the `/docs/` directory and built with Jekyll, one file per component or hook.
+
+All documentation files must include:
+
+- A brief description of the component or hook
+- Usage examples
+- List of props or parameters (required props first, then in alphabetical order)
+- Detailed usage for each prop/parameter (in alphabetical order)
+- Recipes and advanced usage examples if applicable
+
+### Pre-commit Hooks
+
+- Automatic test execution for modified files
+- Prettier formatting check
+- ESLint validation
+- TypeScript compilation
+
+
+### Development Workflow
+```bash
+# Initial setup
+make install         # Install all dependencies
+
+# After making changes
+make build           # Build packages (TypeScript compilation)
+make test            # run unit and e2e tests
+
+# Before pushing changes
+make lint           # Check code quality
+make prettier       # Format code
+```
+
+### Pull Request Process
+
+1. **Target branch**: `next` for features, `master` for bug fixes
+2. **Required checks**:
+   - All tests passing (`make test`)
+   - Linting clean (`make lint`)
+   - Prettier formatted (`make prettier`)
+   - TypeScript compiles (`yarn typecheck`)
+
+3. **Commit Messages**: Clear, descriptive messages focusing on "why"
+   ```
+   fix: Prevent duplicate API calls in useGetList hook
+   feat: Add support for custom row actions in Datagrid
+   docs: Clarify dataProvider response format
+   ```
+4. Use Pull Request Description Template
+
+### Common Make Commands
+```bash
+make                  # Show all available commands
+make install         # Install dependencies
+make build           # Build all packages (CJS + ESM)
+make watch          # Development watch mode
+make test           # Run all tests
+make lint           # Check code quality
+make prettier       # Format code
+make run-simple     # Run simple example
+make run-demo       # Run demo application
+```
+
+### Performance Considerations
+
+- Use `React.memo()` for expensive components
+- Leverage `useMemo()` and `useCallback()` appropriately
```

Comment:
Mention `useEvent` instead of `useCallback`? It may be worth talking about user provided callbacks (don't require users to provide stable references and make it stable in the component/hook ourselves)